app/controllers/accounts_controller.rb

```diff
@@ -132,4 +133,12 @@ def filtered_status_page(params)
       filtered_statuses.paginate_by_max_id(PAGE_SIZE, params[:max_id], params[:since_id]).to_a
     end
   end
+
+  def fields_selection
+    if signed_request_account.present? || !secure_mode_enabled?
```

Comment:
This makes me nervous.  Pleroma's Service actors aren't known via WebFinger.  Will fetches have to be signed by an actor that Mastodon can WebFinger?  If so, is it possible to change that to be pure AP?

Reply:
Did you mean to comment on this particular line? It controls what JSON attributes *we* expose

Separately from this, we *do* require webfinger to process an actor/key, currently. We need to investigate what it would mean to not require that in terms of uniqueness/authority guarantees.

Reply:
does Pleroma at least support webfinger for service actors via URLs (that is, `https:`, not `acct:`)?

Reply:
>we do require webfinger to process an actor/key, currently. We need to investigate what it would mean to not require that in terms of uniqueness/authority guarantees.

wouldn't it be the same? webfinger queries a domain for a resource, but having a direct url should effectively skip the webfinger step, since you already know the resource.

Reply:
Webfinger is host-wide by spec while there is no guarantee the URL path space of a host is not shared by independent entities that could try to impersonate each other... At least I believe that was the original argumentation

Reply:
the security model that makes most sense for the fediverse is that whatever software is running at the root of a domain has control over all keys associated with that domain.  while it is possible that some implementations don't work like this, those implementations don't presently exist (at least not in the mainstream).

thusly, it does not really matter if entities in the path space can impersonate each other, obviously they can, because the software itself controls the keys anyway.

the reason why Pleroma does not publish it's service actors over WebFinger is because we do not want people interacting with them.  they are meant to be used for internal transactions (like relaying presently), and for cases such as handling whatever authorization token is required for fetching remote objects (presently signatures, but ideally bearcaps in the near future) :)

Reply:
as for why i selected this line, it's just the line that brought this thought up about how the signatures will be enforced.

it is pointless for mastodon to try to enforce access control based on what key belonging to a remote host is used to fetch the object: once the remote peer holds the object, it can either respect your wishes, or not.  thusly, i believe it makes most sense to simply never have the metadata exposure in the first place and sign *all* fetches with a designated instance-wide actor (such as `Account.representative`).  this makes the implementation simpler and more robust and doesn't make the implementation try to do things that are ultimately irrelevant anyway.

the question is not whether the software trusts a given user on an instance, but whether the software trusts the instance itself.